core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -872,6 +930,41 @@ public void postProcessing() {
             prepare();
     }
 
+	private void interpolateBridgesAndOrTunnels() {
+		if (!ghStorage.getEncodingManager().supports("generic")) {
+			logger.warn("Elevation interpolation of bridges or tunnels is turned on, but generic flag encoder is not enabled. "
+					+ "Either enable generic flag encoder using graph.flag_encoders=generic,... "
+					+ "or turn off bridge and tunnel interpolation using graph.elevation.interpolate.bridges=false and graph.elevation.interpolate.tunnels=false.");
+			
+		} else {
+			final FlagEncoder genericFlagEncoder = ghStorage.getEncodingManager().getEncoder("generic");
+			if (!(genericFlagEncoder instanceof DataFlagEncoder)) {
+				throw new IllegalStateException("Generic flag encoder equired for elevation interpolation of bridges and tunnels is enabled but does not habe the expected type " + DataFlagEncoder.class.getName() + ".");
+			} else {
```

Comment:
Ok, that's really a matter of taste. I prefer this type if/else blocks, make me think less (like I don't have to check if the if block stops or not). I'd like to leave it as is if it's not that important for you.